pkg/controller/core.oam.dev/v1alpha2/core/components/componentdefinition/componentdefinition_controller.go

```diff
@@ -108,6 +109,11 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 			return ctrl.Result{}, util.PatchCondition(ctx, r, &(componentDefinition),
 				cpv1alpha1.ReconcileError(fmt.Errorf(util.ErrCreateOrUpdateDefinitionRevision, defRev.Name, err)))
 		}
+		klog.InfoS("Successfully update DefinitionRevision", "name", defRev.Name)
+
+		if err := coredef.CleanUpDefinitionRevision(ctx, r.Client, &componentDefinition, r.defRevLimit); err != nil {
```

Comment:
Do you mean the component definition GC doesn't work before?

Reply:
GC does work, but When DefinitionRevision is not a new revision, we also need to enable GC.